src/platform/ESP32/ConnectivityManagerImpl.cpp

```diff
@@ -923,6 +924,14 @@ void ConnectivityManagerImpl::UpdateInternetConnectivityState(void)
                 if (!ip4_addr_isany_val(*netif_ip4_addr(netif)) && !ip4_addr_isany_val(*netif_ip4_gw(netif)))
                 {
                     haveIPv4Conn = true;
+
+                    esp_netif_ip_info_t ipInfo;
+                    if (esp_netif_get_ip_info(esp_netif_get_handle_from_ifkey("WIFI_STA_DEF"), &ipInfo) == ESP_OK)
+                    {
+                        char addrStr[INET_ADDRSTRLEN];
+                        esp_ip4addr_ntoa(&ipInfo.ip, addrStr, sizeof(addrStr));
```

Comment:
should we have some TODO to track things as we move to IPv6?

Reply:
Also filed this issue: https://github.com/project-chip/connectedhomeip/issues/2984